ml3d/datasets/kitti.py

```diff
@@ -0,0 +1,261 @@
+        assert Path(path).exists()
+
+        return np.fromfile(path, dtype=np.float32).reshape(-1, 4)
+
+    @staticmethod
+    def read_label(path):
+        if not Path(path).exists():
+            return None
+
+        with open(path, 'r') as f:
+            lines = f.readlines()
+        objects = [Object3d(line) for line in lines]
+        return objects
+
+    @staticmethod
+    def read_calib(path):
+        assert Path(path).exists()
+
+        with open(path, 'r') as f:
+            lines = f.readlines()
+        obj = lines[2].strip().split(' ')[1:]
+        P2 = np.array(obj, dtype=np.float32)
+
+        obj = lines[3].strip().split(' ')[1:]
+        P3 = np.array(obj, dtype=np.float32)
+
+        obj = lines[4].strip().split(' ')[1:]
+        R0 = np.array(obj, dtype=np.float32)
+
+        obj = lines[5].strip().split(' ')[1:]
+        Tr_velo_to_cam = np.array(obj, dtype=np.float32)
+
+        return {
+            'P2': P2.reshape(3, 4),
+            'P3': P3.reshape(3, 4),
+            'R0': R0.reshape(3, 3),
+            'Tr_velo2cam': Tr_velo_to_cam.reshape(3, 4)
+        }
+
+    def get_split(self, split):
+        return KITTISplit(self, split=split)
+
+    def get_split_list(self, split):
+        cfg = self.cfg
+        dataset_path = cfg.dataset_path
+        file_list = []
+
+        if split in ['train', 'training']:
+            return self.train_files
+            seq_list = cfg.training_split
+        elif split in ['test', 'testing']:
+            return self.test_files
+        elif split in ['val', 'validation']:
+            return val_files
+        elif split in ['all']:
+            return self.train_files + self.val_files + self.test_files
+        else:
+            raise ValueError("Invalid split {}".format(split))
+
+
+class KITTISplit():
+
+    def __init__(self, dataset, split='train'):
+        self.cfg = dataset.cfg
+        path_list = dataset.get_split_list(split)
+        log.info("Found {} pointclouds for {}".format(len(path_list), split))
+
+        self.path_list = path_list
+        self.split = split
+        self.dataset = dataset
+
+    def __len__(self):
+        return len(self.path_list)
+
+    def get_data(self, idx):
+        pc_path = self.path_list[idx]
+        label_path = pc_path.replace('velodyne',
+                                     'label_2').replace('.bin', '.txt')
+        calib_path = label_path.replace('label_2', 'calib')
+
+        pc = self.dataset.read_lidar(pc_path)
+        label = self.dataset.read_label(label_path)
+        calib = self.dataset.read_calib(calib_path)
+
+        data = {
+            'point': pc,
+            'feat': None,
+            'calib': calib,
+            'label': label,
+        }
+
+        return data
+
+    def get_attr(self, idx):
+        pc_path = self.path_list[idx]
+        name = Path(pc_path).name.split('.')[0]
+
+        attr = {'name': name, 'path': pc_path, 'split': self.split}
+        return attr
+
+
+class Object3d(object):
```

Comment:
can we add some docstrings to the new class?

Reply:
Done